### PR TITLE
[Feature] Security Group Target

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000
https://jiracloud.swg.usma.ibm.com:8443/browse/UI-15920

Output from acceptance testing:
Screenshot 2021-04-22 at 1 11 16 PM

Screenshot 2021-04-22 at 3 15 20 PM

Screenshot 2021-05-10 at 11 13 44 AM

```
make testacc TEST=./ibm TESTARGS='-run=TestAccIBMISSecurityGroupTarget_basic'
make testacc TEST=./ibm TESTARGS='-run=TestAccIBMISSecurityGroupTargets_basic'

```

### CODE REVIEW

File: ibm/data_source_ibm_is_security_group_targets.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+			"security_group": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Security group id",
+			},
+
+			"targets": {
+				Type:        schema.TypeList,
+				Description: "List of subnets",
```
Comment (hkantare) [inline ibm/data_source_ibm_is_security_group_targets.go:29]: change the description

File: ibm/data_source_ibm_is_security_group_targets.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"target": {
+							Type:         schema.TypeString,
+							Required:     true,
```
Comment (hkantare) [inline ibm/data_source_ibm_is_security_group_targets.go:36]: Remove required , force new, validate func...targets are computed

File: ibm/data_source_ibm_is_security_group_targets.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+	// Support for pagination
+	start := ""
+	allrecs := []vpcv1.SecurityGroupTargetReferenceIntf{}
+
+	for {
+		listSecurityGroupTargetsOptions := sess.NewListSecurityGroupTargetsOptions(securityGroupID)
+
+		groups, response, err := sess.ListSecurityGroupTargets(listSecurityGroupTargetsOptions)
+		if err != nil {
```
Comment (hkantare) [inline ibm/data_source_ibm_is_security_group_targets.go:83]: add groups == nil else *groups.TotalCount will crash

File: ibm/data_source_ibm_is_security_group_targets.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+	targets := make([]map[string]interface{}, 0)
+	for _, securityGroupTargetReferenceIntf := range allrecs {
+		securityGroupTargetReference := securityGroupTargetReferenceIntf.(*vpcv1.SecurityGroupTargetReference)
+		tr := map[string]interface{}{
+			"name":          *securityGroupTargetReference.Name,
+			"target":        *securityGroupTargetReference.ID,
+			"resource_type": *securityGroupTargetReference.ResourceType,
+		}
+		log.Println("name")
```
Comment (hkantare) [inline ibm/data_source_ibm_is_security_group_targets.go:107]: Remove these log printlns

File: ibm/resource_ibm_is_security_group_target.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+		Read:     resourceIBMISSecurityGroupTargetRead,
+		Delete:   resourceIBMISSecurityGroupTargetDelete,
+		Exists:   resourceIBMISSecurityGroupTargetExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+
+			"security_group": {
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (hkantare) [inline ibm/resource_ibm_is_security_group_target.go:32]: We don't have validate func on security grp id

File: ibm/resource_ibm_is_security_group_target.go [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+	securityGroupID := d.Get("security_group").(string)
+	targetID := d.Get(isSecurityGroupTargetID).(string)
+
+	createSecurityGroupTargetBindingOptions := &vpcv1.CreateSecurityGroupTargetBindingOptions{}
+	createSecurityGroupTargetBindingOptions.SecurityGroupID = &securityGroupID
+	createSecurityGroupTargetBindingOptions.ID = &targetID
+
+	sg, response, err := sess.CreateSecurityGroupTargetBinding(createSecurityGroupTargetBindingOptions)
+	if err != nil {
```
Comment (hkantare) [inline ibm/resource_ibm_is_security_group_target.go:92]: add || sg == nil

File: ibm/resource_ibm_is_security_group_target.go [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error getting Security Group Target : %s\n%s", err, response)
+	}
+
+	target := data.(*vpcv1.SecurityGroupTargetReference)
+	log.Println("*target.Name")
```
Comment (hkantare) [inline ibm/resource_ibm_is_security_group_target.go:129]: Remove this log stmts

File: ibm/resource_ibm_is_security_group_target.go [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+	securityGroupID := parts[0]
+	securityGroupTargetID := parts[1]
+
+	getSecurityGroupTargetOptions := &vpcv1.GetSecurityGroupTargetOptions{
+		SecurityGroupID: &securityGroupID,
+		ID:              &securityGroupTargetID,
+	}
+
+	data, response, err := sess.GetSecurityGroupTarget(getSecurityGroupTargetOptions)
+	if err != nil {
```
Comment (hkantare) [inline ibm/resource_ibm_is_security_group_target.go:120]: || data == nil

**Comment by SunithaGudisagarIBM** [comment]
@hkantare I have incorporated the PR comments, Can you please review this PR..  Thank you

**Pull request merged**